hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -84,6 +84,26 @@ public void beforeMigration(PartitionMigrationEvent event) {
             // 2. Populate non-global partitioned indexes.
             populateIndexes(event, TargetIndexes.NON_GLOBAL);
         }
+
+        flushAndRemoveQueryCaches(event);
+    }
+
+    /**
+     * Flush and remove query cache on this source partition.
+     */
+    private void flushAndRemoveQueryCaches(PartitionMigrationEvent event) {
+        if (event.getMigrationEndpoint() != MigrationEndpoint.SOURCE) {
+            return;
+        }
+
+        QueryCacheContext queryCacheContext = mapServiceContext.getQueryCacheContext();
+        PublisherContext publisherContext = queryCacheContext.getPublisherContext();
+
+        if (event.getMigrationEndpoint() == MigrationEndpoint.SOURCE) {
```

Comment:
Is this check needed? We should already have left the method `if (event.getMigrationEndpoint() != MigrationEndpoint.SOURCE)`.

Reply:
left over. fixed.